src/main/kotlin/no/nav/personbruker/dittnav/eventaggregator/metrics/ProducerNameScrubber.kt

```diff
@@ -0,0 +1,30 @@
+package no.nav.personbruker.dittnav.eventaggregator.metrics
+
+class ProducerNameScrubber(producerAliasesString: String) {
+
+    private val producerNameAliases: Map<String, String> = parseStringAsMap(producerAliasesString)
+
+    val GENERIC_NON_SYSTEM_USER = "non-system-user"
+    val GENERIC_SYSTEM_USER = "unmapped-system-user"
+
+    fun getPublicAlias(producerName: String): String {
+        return if (isNonSystemUser(producerName)) {
+            GENERIC_NON_SYSTEM_USER
+        } else {
+            producerNameAliases[producerName]?: GENERIC_SYSTEM_USER
+        }
+    }
+
+    private fun parseStringAsMap(varString: String): Map<String, String> {
+        return varString.split(",")
+                .map { keyValString -> keyValString.split(":") }
+                .filter { keyValPair -> keyValPair.size == 2 }
+                .map { keyValPair -> keyValPair[0] to keyValPair[1] }
+                .toMap()
+    }
+
+    private fun isNonSystemUser(producer: String): Boolean {
```

Comment:
Er det sannsynlig at vi får eventer fra en NAV-ident?
Vi kunne også snudd litt om på dette, for vi vet at alle systembrukere starter med srv og har en gitt maks lengde.

Reply:
Gikk litt fort i svingene her. Enig med å snu logikken.